cluster/src/main/scala/org/apache/spark/sql/SnappyContextFactory.scala

```diff
@@ -36,11 +36,12 @@ class SnappyContextFactory extends SparkContextFactory {
 
 object SnappyContextFactory {
 
-  private[this] val snappyContextLike =
-    SnappyContext(LeadImpl.getInitializingSparkContext)
+  private[this] val snappySession =
+    new SnappySession(LeadImpl.getInitializingSparkContext)
+
+  protected def newSession(): SnappySession with ContextLike =
+    new SnappySession(snappySession.sparkContext) with ContextLike {
```

Comment:
This problem has been taken care by Spark itself and we don't see multiple tabs now.